src/ui/utils/sanitize.ts

```diff
@@ -0,0 +1,54 @@
+import { Pause, ValueFront } from "protocol/thread";
+import { NodeFront } from "protocol/thread/node";
+import { StyleSheetFront } from "protocol/thread/styleSheet";
+import { StyleFront } from "protocol/thread/style";
+import { RuleFront } from "protocol/thread/rule";
+import { NodeBoundsFront } from "protocol/thread/bounds";
+import { Dispatch, Middleware } from "redux";
+import { UIState } from "ui/state";
+import { UIAction } from "ui/actions";
+
+const forbiddenClasses: Record<string, any> = {
+  Pause,
+  ValueFront,
+  NodeFront,
+  StyleSheetFront,
+  StyleFront,
+  RuleFront,
+  NodeBoundsFront,
+};
+
+const loggedCategories = new Set<string>();
+
+export function sanitize(obj: any, path: string, category: string, logSanitized: boolean): any {
```

Comment:
this is a small nit, but i find it hard to remember what these params do, perhaps this might be cleaner

```js
sanitize(action, { path: "", category: `action[type=${action.type}]`, logSanitized: true });
```

Reply:
One downside with Typescript is that it makes writing a function that takes its parameters as an object a bit cumbersome:
```
function sanitize(obj: any, { path, category, logSanitized }: { path: string; category: string; logSanitized: boolean }) {...}
```
So I tend to avoid it if there aren't many callers, because it doesn't seem worth making the function declaration less readable. After all, Typescript makes it possible to peek at the function declaration with one keystroke, so it's not necessary to make the call more explicit.